docs/proposals/registry/registry-structure.md

```diff
@@ -36,6 +36,36 @@ The following requirements are imposed on the devfile registry repository:
 2) Each devfile stack folder must contain a `devfile.yaml`. Other files such as vsx plugins, stack logos, etc. can be included as needed. 
 3) Any build tools for the registry (such as scripts or Dockerfiles) should not be located at the top-level of the stack, and should reside in their own folder.
 
+Optional file to support stacks/samples from other repositories:
+
+A file called `extraDevfileEntries.yaml` can be added under the root of the devfile registry source repository to add samples and stacks from other repositories to the registry. This file contains the location information on where the extra samples and stacks can be found.
+
+Sample `extraDevfileEntries.yaml`:
+```
+schemaVersion: 1.0.0
+samples:
+  - name: nodejs-basic
+    displayName: Basic NodeJS
+    description: A simple Hello World application
+    icon: nodejsIcon.svg
+    tags: ["NodeJS", "Express"]
+    projectType: nodejs
+    language: nodejs
+    git:
+      remotes:
+        origin: https://github.com/redhat-developer/devfile-sample
```

Comment:
Loooks good to me, except maybe one ques that pops into my head

what is the expected outcome(index) if git has multiple remotes.

```yaml
git:
    remotes:
        origin: https://github.com/redhat-developer/devfile-sample
        my-fork: https://github.com/my-repo/devfile-sample
```

Reply:
The struct of remotes is a map
```
type Git struct {
	Remotes map[string]string `yaml:"remotes,omitempty" json:"remotes,omitempty"`
}
```
which means it is a map with multiple entries if the git has multiple remotes, so it's similar to your example above

Reply:
I understand, but what is the expected index.. Will we have two index entries pointing to both the remote repositories? How does registry-support know which repo to choose in this case to point the sample/stack 🤔

Reply:
For samples and stacks from other repos (resources not store in the registry), we only display the metadata to consumers from the index and they decide how to use them. Also I rethink this, given the `git` field is only for samples and stacks from other repos , then the relationship should be one sample/stack per repo, it's different from one generic stack contains multiple starter projects and multiple repos.

Reply:
For sample projects, I don't think it should have multiple remote (same reason as starterProjects). Both samples and starterProjects are only being cloned one and then it gets detached from the original, i.e. no git remote is setup. Therefore, it doesn't make sense to have multiple remotes. We should restrict that use case.

Reply:
To restrict it, I think we add it as part of the validation code in the index generator.